src/widgets/ot_pulltorefresh.eliom

```diff
@@ -0,0 +1,261 @@
+[%%shared.start]
+
+[%%client 
+  open Eliom_content.Html
+]
+
+open Eliom_content.Html.D
+
+[%%client
+
+  module type PULLTOREFRESH = sig
+    val dragThreshold : float 
+    val moveCount : int
+    val headContainerHeight : int
+    val pullDownIcon : Html_types.div  Eliom_content.Html.D.elt
+    val loadingIcon : Html_types.div  Eliom_content.Html.D.elt
+    val successIcon : Html_types.div  Eliom_content.Html.D.elt
+    val failureIcon : Html_types.div  Eliom_content.Html.D.elt
+    val pullText :  Html_types.span  Eliom_content.Html.D.elt 
+    val container :  Html_types.div  Eliom_content.Html.D.elt
+    val pullDownText : string
+    val releaseText : string
+    val loadingText : string
+    val successText : string
+    val failureText : string
+    val rotateGradually : bool
+    val blockPullIcon : bool
+    val afterPull : unit -> bool Lwt.t
+  end
+
+  module Make(Elt:PULLTOREFRESH) = struct
+    let dragThreshold = Elt.dragThreshold 
+    let moveCount  = min (max 100 Elt.moveCount) 500
+    let dragStart = ref (-1) 
+    let percentage = ref 0. 
+    let joinRefreshFlag = ref false 
+    let refreshFlag = ref false 
+    let pullText = To_dom.of_element Elt.pullText 
+    let container = Elt.container
+    let js_container = To_dom.of_element container
+
+    let show =
+      let icon_list = 
+        [Elt.pullDownIcon; Elt.loadingIcon; Elt.successIcon; Elt.failureIcon]
+      in
+      fun elt ->
+        List.iter (fun x -> 
+          if x == elt 
+          then Manip.Class.remove x "ot-display-none" 
+          else Manip.Class.add x "ot-display-none") 
+          icon_list 
```

Comment:
I don't know if you have access to:
https://github.com/besport/bs/wiki/Coding-style#avoid-brain-stack-overflow
If not nthe rule of thumb is: write `l |> List.map f` in whenever `l` is simple and `f` is complex, but stick with `List.map f l` when `f` is simple and `l` is complex.